### PR TITLE
actions: added libaom to ffmpeg to catch symbol conflicts and fix macOS symbol conflicts

### PR DESCRIPTION
**Pull request by ccccheung**
# Description
updates ffmpeg CI to be built with libaom enabled to catch symbol conflicts. 
fixes macOS symbol conflicts
# Issue

# Author(s)
Colton Cheung
Christopher Degawa

# Performance impact

- [ ] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [x] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [ ] other
- [x] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

File: .github/workflows/patches.yml [hunk 1]
```diff
@@ -70,12 +70,19 @@ jobs:
           repository: FFmpeg/FFmpeg
           ref: release/4.2
           path: ffmpeg-release-4.2
+      - name: Clone libaom
+        run: git clone https://aomedia.googlesource.com/aom
+      - name: Configure libaom
+        run: |
+          [ -d ~/.ccache ] && sudo chown -R "$USER": ~/.ccache
+          cmake -S aom -B aom/build-static -DCMAKE_BUILD_TYPE=Debug -DBUILD_SHARED_LIBS=OFF -DENABLE_TESTS=0
```
Comment (1480c1) [inline .github/workflows/patches.yml:78]:
```suggestion
          cmake -S aom -B aom/build-static -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DENABLE_TESTS=0
```
Could you make this release to match the svt-av1 make above?

File: .github/workflows/patches.yml [hunk 2]
```diff
@@ -70,12 +70,19 @@ jobs:
...
+          sudo make -C aom/build-static -j "$(getconf _NPROCESSORS_ONLN 2>/dev/null || sysctl -n hw.ncpu)" install
```
Comment (1480c1) [inline .github/workflows/patches.yml:79]:
```suggestion
          sudo -E make -C aom/build-static -j "$(getconf _NPROCESSORS_ONLN 2>/dev/null || sysctl -n hw.ncpu)" install
```

use `sudo -E` to make sure cache works and the cache goes to the correct directory

**Comment by 1480c1** [comment]
seems there are still some symbols conflicting for macOS

**Comment by hassount** [comment]
@ccccheung @1480c1 is this ready to go ?

**Comment by 1480c1** [comment]
I'm not sure since I think we may need to merge https://github.com/ccccheung/SVT-AV1/pull/2 since the codec name was changed upstream to be consistent, although, we can merge it afterward.

**Comment by ccccheung** [comment]
@hassount this PR is good to merge

**Pull request merged**